src/libraries/System.Runtime/tests/TrimmingTests/InvariantGlobalizationTrue.cs

```diff
@@ -13,18 +12,32 @@ class Program
 {
     static int Main(string[] args)
     {
-        // since we are using Invariant GlobalizationMode = true, setting the culture doesn't matter.
-        // The app will always use Invariant mode, so even in the Turkish culture, 'i' ToUpper will be "I"
-        Thread.CurrentThread.CurrentCulture = new CultureInfo("tr-TR");
+        // Ensure the internal GlobalizationMode class is trimmed correctly
+        Type globalizationMode = GetCoreLibType("System.Globalization.GlobalizationMode");
+        const BindingFlags allStatics = BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Static;
+        bool predefinedCulturesOnly = (bool) globalizationMode.GetProperty("PredefinedCulturesOnly", allStatics).GetValue(null);
+
+        try
+        {
+            CultureInfo.CurrentCulture = new CultureInfo("tr-TR");
+            if (predefinedCulturesOnly)
+            {
+                return -1; // we expect new CultureInfo("tr-TR") to throw.
+            }
+        }
+        catch (CultureNotFoundException)
+        {
+            if (!predefinedCulturesOnly)
```

Comment:
When is this ever false? It always gets replaced with true when the app is trimmed.

Reply:
I was not sure if we run into any situation forcing it to false as we saw WASM was forcing it to true. anyway, if you are sure this will never be false then I'll modify the test according to that.

Reply:
I have updated the test according to your assumption and let's see how it goes in CI.

Reply:
> anyway, if you are sure this will never be false then I'll modify the test according to that.

You are explicitly passing `--feature System.Globalization.PredefinedCulturesOnly true` into the linker for this test. That tells the linker to replace `GlobalizationMode.PredefinedCulturesOnly`'s code to just return a hard-coded `true`. It can never be `false` in this test because of this.